invenio_sipstore/tasks.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Background tasks for sipstore."""
+
+from __future__ import absolute_import, print_function
+
+import os
+
+from datetime import datetime
+
+from celery import shared_task
+from fs.opener import opener
+from invenio_db import db
+
+from .archivers import BagItArchiver
+from .models import SIP
+
+
+@shared_task
+def archive_sip(sip_id, ):
+    """Celery task to create a bagit file from a sip package."""
+    sip = SIP.query.get(sip_id)
+
+    archival_start = datetime.now()
+
+    location = os.path.join('basepath', str(sip_id),
```

Comment:
I think it is better if the BagItArchiver receives only the FS where is going to make the mess. The creation of the FS is not its responsibility.

However, I agree that the ``basepath`` is currently hard coded and needs a better idea. About its configuration, this is a Celery task, what about a new parameter in the task?

```python
@shared_task
def archive_sip(sip_id, location):
```

Is this part going to suffer changes? I mean, @krzysztof commented me to add a new parameter to the model (``archive_uri``) in order to know the path. It is not currently implemented because you know where is archived thanks to the location pattern ('basepath/< sip_id >/< archival_at >/', but should I add it?